src/Devices/OWSVerificationStateSyncMessage.m

```diff
@@ -0,0 +1,102 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+#import "OWSVerificationStateSyncMessage.h"
+#import "OWSSignalServiceProtos.pb.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface OWSVerificationStateTuple : NSObject
+
+@property (nonatomic) OWSVerificationState verificationState;
+@property (nonatomic) NSData *identityKey;
+@property (nonatomic) NSString *recipientId;
+
+@end
+
+#pragma mark -
+
+@implementation OWSVerificationStateTuple
+
+@end
+
+#pragma mark -
+
+@interface OWSVerificationStateSyncMessage ()
+
+@property (nonatomic, readonly) NSMutableArray<OWSVerificationStateTuple *> *tuples;
+
+@end
+
+#pragma mark -
+
+@implementation OWSVerificationStateSyncMessage
+
+- (instancetype)init
+{
+    self = [super init];
+    if (!self) {
+        return self;
+    }
+    
+    _tuples = [NSMutableArray new];
+    
+    return self;
+}
+
+- (void)addVerificationState:(OWSVerificationState)verificationState
+                 identityKey:(NSData *)identityKey
+                 recipientId:(NSString *)recipientId
+{
+    OWSAssert(identityKey.length > 0);
+    OWSAssert(recipientId.length > 0);
+    OWSAssert(self.tuples);
+
+    OWSVerificationStateTuple *tuple = [OWSVerificationStateTuple new];
```

Comment:
If we did that, I would always be worried about it accidentally getting saved or something.